staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -658,9 +662,11 @@ func (s *store) Stats(ctx context.Context) (stats storage.Stats, err error) {
 	}, nil
 }
 
-func (s *store) SetKeysFunc(keys storage.KeysFunc) {
-	if s.stats != nil {
-		s.stats.SetKeysFunc(keys)
+func (s *store) EnableResourceSizeEstimation(getKeys storage.KeysFunc) {
```

Comment:
does this do the right thing if called repeatedly, or called with nil after already being set?

Reply:
Added protection from being called with nil. 

As for multiple calls, I think taking first is good enough.

Reply:
That's a good point, need to think more about preventing bugs as code evolves. Thanks.